src/annotator/annotation-sync.js

```diff
@@ -1,176 +1,116 @@
-        }
-        return parsedAnnotations;
-      };
-    })(this)
-  );
-  return this;
-};
-
-// Handlers for messages arriving through a channel
-AnnotationSync.prototype._channelListeners = {
-  deleteAnnotation: function (body, cb) {
-    const annotation = this._parse(body);
-    delete this.cache[annotation.$tag];
-    this._emit('annotationDeleted', annotation);
-    cb(null, this._format(annotation));
-  },
-  loadAnnotations: function (bodies, cb) {
-    const annotations = function () {
-      let i;
-      const parsedAnnotations = [];
 
-      for (i = 0; i < bodies.length; i++) {
-        parsedAnnotations.push(this._parse(bodies[i]));
+    // Relay events from annotator to sidebar.
+    this._on('beforeAnnotationCreated', annotation => {
+      if (annotation.$tag) {
+        return;
       }
-      return parsedAnnotations;
-    }.call(this);
-    this._emit('annotationsLoaded', annotations);
-    return cb(null, annotations);
-  },
-};
+      this.bridge.call('beforeCreateAnnotation', this._format(annotation));
+    });
+  }
 
-// Handlers for events coming from this frame, to send them across the channel
-AnnotationSync.prototype._eventListeners = {
-  beforeAnnotationCreated: function (annotation) {
-    if (annotation.$tag) {
-      return undefined;
-    }
-    return this._mkCallRemotelyAndParseResults('beforeCreateAnnotation')(
-      annotation
+  /**
+   * Relay updated annotations from the annotator to the sidebar.
+   *
+   * This is called for example after annotations are anchored to notify the
+   * sidebar about the current anchoring status.
+   */
+  sync(annotations) {
+    this.bridge.call(
+      'sync',
+      annotations.map(ann => this._format(ann))
     );
-  },
-};
-
-AnnotationSync.prototype._mkCallRemotelyAndParseResults = function (
-  method,
-  callBack
-) {
-  return (function (_this) {
-    return function (annotation) {
-      // Wrap the callback function to first parse returned items
-      const wrappedCallback = function (failure, results) {
-        if (failure === null) {
-          _this._parseResults(results);
-        }
-        if (typeof callBack === 'function') {
-          callBack(failure, results);
-        }
-      };
-      // Call the remote method
-      _this.bridge.call(method, _this._format(annotation), wrappedCallback);
-    };
-  })(this);
-};
-
-// Parse returned message bodies to update cache with any changes made remotely
-AnnotationSync.prototype._parseResults = function (results) {
-  let bodies;
-  let body;
-  let i;
-  let j;
+  }
 
-  for (i = 0; i < results.length; i++) {
-    bodies = results[i];
-    bodies = [].concat(bodies);
-    for (j = 0; j < bodies.length; j++) {
-      body = bodies[j];
-      if (body !== null) {
-        this._parse(body);
-      }
+  /**
+   * Assign a non-enumerable tag to objects which cross the bridge.
+   * This tag is used to identify the objects between message.
+   *
+   * @param {string} [tag]
+   */
+  _tag(ann, tag) {
```

Comment:
Can we use fully-spelled-out `annotation` here instead of `ann`? Most other references in this module spell it out, and it's easier for later grepping (as `ann` could be short for `annotation` or `annotator`).

Reply:
We do use `ann` as a shorthand for `annotation` quite widely within various parts of Hypothesis, but I agree the full version would be more consistent here.